rasa/nlu/selectors/response_selector.py

```diff
@@ -338,21 +349,31 @@ def _full_response(self, label: Dict[Text, Any]) -> Optional[Dict[Text, Any]]:
             the text but also buttons, images, ...
         """
         for key, responses in self.responses.items():
-            for response in responses:
-                if hash(response.get(TEXT, "")) == label.get("id"):
-                    return response
-        return None
+            if self.train_on_text:
+                for response in responses:
+                    if hash(response.get(TEXT, "")) == label.get("id"):
+                        return key, responses
+            else:
+                if hash(key) == label.get("id"):
+                    return key, responses
+
+        # If none of the responses match, then go to default response where response key
+        # and the response text are the same.
+        return label.get("name"), [{"text": label.get("name")}]
 
     def process(self, message: Message, **kwargs: Any) -> None:
         """Return the most likely response and its similarity to the input."""
 
         out = self._predict(message)
         label, label_ranking = self._predict_label(out)
-        retrieval_intent_name = self.retrieval_intent_mapping.get(label.get("name"))
+
+        label_retrieval_intent, label_responses = self._full_response(label)
 
         for ranking in label_ranking:
-            ranking["full_retrieval_intent"] = self.retrieval_intent_mapping.get(
-                ranking.get("name")
+            ranking["full_retrieval_intent"] = (
```

Comment:
is a `full_retrieval_intent ` the same as `example.get_combined_intent_response_key`? if so please unify the naming